Docs update (#81)
gytis-ivaskevicius/flake-utils-plus: README.md

[![Discord](https://img.shields.io/discord/591914197219016707.svg?label=&logo=discord&logoColor=ffffff&color=7389D8&labelColor=6A7EC2)](https://discord.com/invite/RbvHtGa)

Need help? Create an issue or ping @Gytis#0001 in the above Discord Server.

# What is this flake?

Flake-utils-plus exposes a library abstraction to *painlessly* generate NixOS flake configurations.

The biggest design goal is to keep down the fluff. The library is meant to be easy to understand and use. It aims to be far simpler than frameworks such as DevOS (previously called nixflk).

# Features of flake #

> FUP provides two main features (visible from `flake.nix`):

Reviewer: ```suggestion
FUP provides a few main features (visible from `flake.nix`):
```

- `nix.generateRegistryFromInputs` - Generates `nix.nixPath`/`nix.registry` from flake `inputs`.
- `lib.mkFlake { ... }` - Generates a flake using FUP magic.

> - `lib.exportModules [ ./a.nix ./b.nix ]` - Generates modules attribute which looks like this `{ a = import ./a.nix; b = import ./b.nix; }`.

Reviewer: ```suggestion
- `lib.exportModules [ ./a.nix ./b.nix ]` - Generates module attribute which look like this `{ a = import ./a.nix; b = import ./b.nix; }`.
```

- `lib.exportOverlays channels` - Exports all overlays from channels as an appropriately namespaced attribute set. Users can instantiate with their nixpkgs version.

> - `lib.outputsBuilder.packages channels pkgs` - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Users can use your cache.

Reviewer: ```suggestion
- `lib.outputsBuilder.packages channels pkgs` - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Unlike overlays, these packages are consistent across flakes allowing them to be cached.
```

> - `lib.exportModules [ ./a.nix ./b.nix ]` - Generates module attribute which look like this `{ a = import ./a.nix; b = import ./b.nix; }`.
> - `lib.exportOverlays channels` - Exports all overlays from channels as an appropriately namespaced attribute set. Users can instantiate with their nixpkgs version.
> - `lib.outputsBuilder.packages channels pkgs` - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Unlike overlays, these packages are consistent across flakes allowing them to be cached.

Author: Do you mean something like this?
```suggestion
- [`lib.exportModules [ ./a.nix ./b.nix ]`](./lib/exportModules.nix) - Generates module attribute which look like this `{ a = import ./a.nix; b = import ./b.nix; }`.
- [`lib.exportOverlays channels`](./lib/exportOverlays.nix) - Exports all overlays from channels as an appropriately namespaced attribute set. Users can instantiate with their nixpkgs version.
- [`lib.outputsBuilder.packages channels pkgs`](./lib/outputsBuilder.nix) - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Unlike overlays, these packages are consistent across flakes allowing them to be cached.
```

> - `lib.outputsBuilder.packages channels pkgs` - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Unlike overlays, these packages are consistent across flakes allowing them to be cached.

Reviewer 2: yeah, something like that. Tho I don't think that URL part will work the way you did it

# How to use this flake #

* [Example flake with all available attributes](https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/examples/fully-featured/flake.nix)

> TODO: we should add it back.

Author: Alright then.

  #############################
  ### flake outputs builder ###
  #############################

>   # THESE ARE GUESSES - pls review

Reviewer 2: ```suggestion
```
